edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -88,7 +100,7 @@ void Callback(object _state)
                     ? this.maxRenewAfter
                     : renewAfter;
                 this.logger.LogDebug("Scheduling server certificate renewal timer for {0}.", DateTime.UtcNow.Add(clamped).ToString("o"));
-                this.timer.Change(clamped, Timeout.InfiniteTimeSpan);
+                this.timer.Change(clamped, this.maxCheckCertExpiryAfter);
```

Comment:
shouldn't it change here to lowest value between clamped and this.maxCheckCertExpiryAfter and keep period as InfiniteTimeSpan?